src/containers/EditListingPage/EditListingWizard/EditListingDetailsPanel/EditListingDetailsForm.js

```diff
@@ -115,6 +115,111 @@ const FieldSelectListingType = props => {
   );
 };
 
+// Finds the correct subcategory within the given categories array based on the provided categoryIdToFind.
+const findCorrectSubcategory = (categories, categoryIdToFind) => {
+  return categories?.find(category => category.id === categoryIdToFind);
+};
+
+/**
+ * Recursively render subcategory field inputs if there are subcategories available.
+ * This function calls itself with updated props to render nested category fields.
+ * The select field is used for choosing a category or subcategory.
+ */
+const CategoryField = props => {
+  const { currentCategoryOptions, level, values, prefix, handleCategoryChange, intl } = props;
+
+  const currentCategoryKey = `${prefix}${level}`;
+
+  const nextSubcategories = findCorrectSubcategory(
+    currentCategoryOptions,
+    values[`${prefix}${level}`]
+  );
+
+  return (
+    <>
+      {level === 1 || values[currentCategoryKey] || values[`${prefix}${level - 1}`]?.length > 0 ? (
```

Comment:
I'm not sure what this is supposed to do:
```js
values[`${prefix}${level - 1}`]?.length > 0
```

How about adding these above `return (`:
```js
const currentCategoryValue = values[currentCategoryKey];
const isCategoryLevelSet = typeof currentCategoryValue !== 'undefined';
const isValidCategoryValue = !!categoryConfig;
```

```suggestion
      {isCategoryLevelSet && isValidCategoryValue ? (
```

Reply:
@Gnito 
```values[`${prefix}${level - 1}`]?.length > 0``` This checks if the parent category is selected. A field is only rendered if it is a top-level field `(level === 1)`, if the user has selected a value `values[currentCategoryKey]` or if the parent category has a value.

Would this work:

```
  const isCurrentCategoryValueSelected = values[currentCategoryKey];
  const isTopLevelCategory = level === 1;
  const isParentCategorySelected = values[`${prefix}${level - 1}`]?.length > 0;
  return (
    <>
      {isTopLevelCategory || isCurrentCategoryValueSelected || isParentCategorySelected ? (
```